app/lib/feature_flags.rb

```diff
@@ -22,19 +27,24 @@ def enabled?(flag, session_data = nil, without_session_data: false)
         raise "Pass in session_data or set without_session_data to true"
       end
 
+      #  if the flag exists in the session_data use that setting
       if session_data && session_data["feature_flags"]&.key?(flag.to_s)
         return session_data["feature_flags"][flag.to_s]
       end
 
-      if overrideable?
-        override = FeatureFlagOverride.find_by(key: flag)
-        if override
-          return override.value
-        end
-      end
+      possible_override = overrideable_flag(flag)
+      return possible_override unless possible_override.nil?
 
       if STATIC_FLAGS.key?(flag)
-        ENV["#{flag.to_s.upcase}_FEATURE_FLAG"]&.casecmp("enabled")&.zero? || STATIC_FLAGS.fetch(flag).fetch(:default)
+        # if the flag is a valid flag and of type `session` but does not
+        # exist in the session_data we return the default value for that flag
+        if STATIC_FLAGS[flag][:type] == "session"
+          STATIC_FLAGS[flag][:default]
```

Comment:
so do we never look at the env_var anymore for session flags (when it is not in session_data/there is no session_data)? We only ever use the default?

Reply:
How would we then manage session flags being switched on or off across multiple envs? Perhaps I am misunderstanding.

Reply:
>never look at the env_var anymore for session flags
correct. If a check starts without a new feature and we enable the env_var to release the feature, we would be passing 'enabled' to the check, which would then break.
so the only time we should check the env_var (for session flags) is when the check starts using the method `session_flags`

Reply:
Setting env_vars across multiple envs, or even mutiple UATs will work because the env_var is used to set the initial state of the check being undertaken. These flags are then set on a per check basis and not a per environment depending on the configuration of the deployment at the time the check started, so we can change env_vars between environments or even between checks on the same environment, and the check will use the inital configuration as its flags.

Reply:
Yeah I think I'm overthinking it. I was thinking, is there a situation where a session flag might not have session data passed in...(where we set `without_session_data: true`) but tbh I don't _think_ those should/would be session flags. 

I'm thinking for places like the start page where we don't have a session. But I wonder if we might have grouped a bunch changes together under a session flag, because they're all part of the same feature, but not pass the session in, in places where we don't have it. Probably overthinking it but we will just need to be a bit careful with our flag setup...

Reply:
It feels like a very edgy case nasty bug lol